### ISSUE TITLE
Nothing wrong.

### ISSUE THREAD

**Issue by NZ-Liam-Zhong** [uclaml/SPIN#41]
In your trianer.py, there is a :<br>
 pi_logratios = policy_real_logps - policy_generated_logps
        ref_logratios = opponent_real_logps - opponent_generated_logps

        if reference_free:
            ref_logratios = 0

        logits = pi_logratios - ref_logratios

It is different with SPIN loss.<br>
the spin loss should be <br>
 logits = pi_logratios + ref_logratios

**Comment by NZ-Liam-Zhong** [comment]
![Image](https://github.com/user-attachments/assets/49c163a1-5c21-49ab-becc-e692a0f27ec7)
It should be:
pi_logratios = policy_real_logps -  opponent_generated_logps
ref_logratios = opponent_real_logps - policy_generated_logps
logits = pi_logratios + ref_logratios

**Comment by NZ-Liam-Zhong** [comment]
I am wrong, sorry
